### PR TITLE
MDO-Enrichment-Update-17-11

### PR DESCRIPTION
**Pull request by rushriva**
Required items, please complete
   
   Change(s):
   - PR is raised after https://github.com/Azure/Azure-Sentinel/pull/6605 PR has merge conflicts

   Reason for Change(s):
   - Enriching Microsoft 365 defender solution

   Version Updated:
   - NA

   Testing Completed:
   - yes

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by rushriva** [comment]
@v-spadarthi  - Please assign this to @devikamehra

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json [hunk 1]
```diff
@@ -0,0 +1,1317 @@
...
+            },
+            "customWidth": "25",
+            "name": "query - 0 - Copy - Copy"
+          },
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "EmailEvents\r\n|where DetectionMethods has_any ('spoof','impersonation')\r\n| project DT=parse_json(DetectionMethods), EmailDirection, SenderFromAddress, AdditionalFields \r\n| evaluate bag_unpack(DT) \r\n| summarize count()by tostring(Phish[0]) ",
+              "size": 0,
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:194]: Can we change the height to Tiny. The other two queries in the same line are at the same height.

Just a suggestion, if deemed appropriate, let's include all the count KQLs in a single query.

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json [hunk 2]
```diff
@@ -0,0 +1,1317 @@
...
+                    }
+                  },
+                  {
+                    "columnMatch": "NetworkMessageIds",
+                    "formatter": 5
+                  }
+                ]
+              }
+            },
+            "customWidth": "35",
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:540]: Can we change this to 50 and make the adjacent query also of the same width.

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json [hunk 3]
```diff
@@ -0,0 +1,1317 @@
...
+      "type": 12,
+      "content": {
+        "version": "NotebookGroup/1.0",
+        "groupType": "editable",
+        "items": [
+          {
+            "type": 3,
+            "content": {
+              "version": "KqlItem/1.0",
+              "query": "EmailEvents \r\n| project Timestamp, AR=parse_json(AuthenticationDetails) , NetworkMessageId, EmailDirection, SenderFromAddress, ThreatTypes, DetectionMethods   \r\n| evaluate bag_unpack(AR) \r\n| where DMARC == \"fail\" \r\n| summarize count() by bin(Timestamp, 1d) ",
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:763]: Change the comparison to =~
Applicable to all the queries in this section.

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json [hunk 4]
```diff
@@ -0,0 +1,1317 @@
...
+                    }
+                  },
+                  {
+                    "columnMatch": "NetworkMessageIds",
+                    "formatter": 5
+                  }
+                ]
+              }
+            },
+            "customWidth": "30",
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:1297]: Expand this to 50

File: Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+tactics:
+- Initial access
+query: |
+  let cutoff = 5;
+  EmailEvents
+  | where ThreatTypes has "Malware" or ThreatTypes has "Phish" 
+  | summarize count() by SenderIPv4 
+  | where count_ > cutoff // Arbitrary cutoff, increase or decrease as needed 
+  | join EmailEvents on SenderIPv4  
+  | where DeliveryAction == "Delivered"
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml:18]: Can we change this too =~ for case sensitivity.

File: Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  dataTypes:
+  - EmailEvents
+tactics:
+- Initial access
+query: |
+  EmailEvents 
+  | where TimeGenerated > ago (1d) and DetectionMethods has "spoof" and SenderFromDomain has "contoso.com"  // contoso.com is provided for refrence, this can be replaced by specific domain value
+  | project TimeGenerated, AR=parse_json(AuthenticationDetails) , NetworkMessageId, EmailDirection, Subject, SenderFromAddress, SenderIPv4,ThreatTypes,   DetectionMethods, ThreatNames  
+  | evaluate bag_unpack(AR)  
+  | where SPF == "fail" or DMARC == "fail" or DKIM == "fail" or CompAuth == "fail"
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml:16]: == --> =~

File: Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+id: cdac93ef-56c0-45bf-9e7f-9cbf0ad06123
+name: Determines emails sent by top malicious/bad IP addresses.
```
Comment (devikamehra) [inline Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml:2]: Change the name of the query.

**Review by devikamehra — CHANGES_REQUESTED** [review]
Changes are required

**Comment by v-spadarthi** [comment]
@rushriva : Please resolve the below comments.
After deploying the workbook we are getting below issues please fix it
image
By Authentication section we are getting below ,please fix it 
image
Please add the workbook meta data for above workbook in below path https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json (hunk 1 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:194]: completed

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json (hunk 2 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:540]: done

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json (hunk 3 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:763]: done

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json (hunk 4 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json:1297]: done

File: Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml (hunk 5 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml:18]: done

File: Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml (hunk 6 above)
Comment (rushriva) [reply Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml:16]: done

File: Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json (hunk 2 above)
Comment (devikamehra) [reply Solutions/Microsoft 365 Defender/Workbooks/MDO-Workbook.json]: Not updated

**Review by devikamehra — APPROVED** [review]

**Comment by rushriva** [comment]
> @rushriva : Please resolve the below comments. After deploying the workbook we are getting below issues please fix it image By Authentication section we are getting below ,please fix it image Please add the workbook meta data for above workbook in below path https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json

Resolved - @v-spadarthi  comments resolved, Please proceed with packaging

image

**Comment by v-spadarthi** [comment]
Workbook is looking good 
image
@rushriva : Please add Workbook metadata missing please add in below path 
 https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json

**Comment by v-spadarthi** [comment]
Workbook looking good 
image
Workbook meta data raised for another [PR](https://github.com/Azure/Azure-Sentinel/pull/6694)

**Pull request merged**